python/src/rest_api/aac_rest_app.py

```diff
@@ -0,0 +1,531 @@
+    aac_command = aac_commands_by_name.get(command_request.name)
+
+    if aac_command is not None:
+        aac_command_argument_names = [arg.name for arg in aac_command.arguments]
+        arguments = [arg for arg in command_request.arguments if arg not in aac_command_argument_names]
+
+        try:
+            result = aac_command.callback(*(arguments or []))
+            success = result.status_code == ExecutionStatus.SUCCESS
+            result_message = f"{result.name}: {result.status_code.name.lower()}\n\n{result.get_messages_as_string()}"
+        except Exception as error:
+            success = False
+            result_message = f"{result.name}: failure\n\n{error}"
+        finally:
+            return CommandResponseModel(command_name=aac_command.name, result_message=result_message, success=success)
+    else:
+        _report_error_response(
+            HTTPStatus.NOT_FOUND,
+            f"Command name {command_request.name} not found in the list of available commands: {list(aac_commands_by_name.keys())}.",
+        )
+
+
+def _get_available_files_in_workspace() -> set[AaCFile]:
+    """
+    Get the available AaC files in the workspace sans files already in the context.
+
+    Args:
+        active_context (LanguageContext): The active language context.
+
+    Returns:
+        A set containing available AaC files in workspace.
+    """
+    aac_files_in_context = set(_get_files_in_context())
+    aac_files_in_workspace = set(find_aac_files(WORKSPACE_DIR))
+
+    return aac_files_in_workspace.difference(aac_files_in_context)
+
+
+async def refresh_available_files_in_workspace(context: LanguageContext) -> None:
+    """
+    Used to refresh the available files. Used in async since it takes too long for being used in request-response flow.
+
+    Args:
+        active_context (LanguageContext): The active Language Context.
+    """
+    global AVAILABLE_AAC_FILES
+    global active_context
+    active_context = context
+    AVAILABLE_AAC_FILES = list(_get_available_files_in_workspace())
+
+    # Update the active context with any missing files
+    files_in_context = {file.uri for file in _get_files_in_context()}
+    available_files = {file.uri for file in AVAILABLE_AAC_FILES}
+    missing_files = available_files.difference(files_in_context)
+    try:
+        definition_lists_from_missing_files = [parse(file_uri) for file_uri in missing_files]
+    except ParserError as error:
+        raise ParserError(error.source, error.errors) from None
+    else:
+        definitions_to_add = {definition.name: definition for definition_list in definition_lists_from_missing_files for definition in definition_list}
+        parser = DefinitionParser()
+        parser.load_definitions(active_context, list(definitions_to_add.values()))
+        # active_context.add_definitions_to_context(list(definitions_to_add.values()))
+
+
+def _report_error_response(code: HTTPStatus, error: str):
```

Comment:
Still need docstrings

Reply:
So, for error/exception handling methods, rather than a `Returns` entry to the docstring, you would have a `Raises` entry. 

https://stackoverflow.com/questions/34160968/python-docstring-raise-vs-raises